module/msg_smt/src/mod_msg_smt.c

```diff
@@ -0,0 +1,477 @@
+/*
+ * Arm SCP/MCP Software
+ * Copyright (c) 2022, Linaro Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Description:
+ *     msg buffer device driver.
+ */
+
+#include <stdbool.h>
+#include <string.h>
+#include <fwk_assert.h>
+#include <fwk_interrupt.h>
+#include <fwk_log.h>
+#include <fwk_mm.h>
+#include <fwk_macros.h>
+#include <fwk_module.h>
+#include <fwk_module_idx.h>
+#include <fwk_status.h>
+#include <mod_scmi.h>
+#include <mod_msg_smt.h>
+
+struct __attribute((packed)) mod_msg_smt_memory {
+    uint32_t message_header;
+    uint32_t payload[];
+};
+
+struct smt_channel_ctx {
+    /* Channel identifier */
+    fwk_id_t id;
+
+    /* Channel configuration data */
+    struct mod_msg_smt_channel_config *config;
+
+    /* Channel read and write cache memory areas */
+    struct mod_msg_smt_memory *in;
+    struct mod_msg_smt_memory *out;
+    size_t in_len;
+    size_t out_len;
+
+    /* Message processing in progrees flag */
+    volatile bool locked;
+
+    /* Maximum payload size of the channel */
+    size_t max_payload_size;
+
+    /* Driver entity identifier */
+    fwk_id_t driver_id;
+
+    /* SCMI module service bound to the channel */
+    fwk_id_t scmi_service_id;
+
+    /* Driver API */
+    struct mod_msg_smt_driver_ouput_api *driver_api;
+
+    /* SCMI service API */
+    struct mod_scmi_from_transport_api *scmi_api;
+
+    /* Flag indicating the mailbox is ready */
+    bool msg_smt_mailbox_ready;
+};
+
+struct smt_ctx {
+    /* Table of channel contexts */
+    struct smt_channel_ctx *channel_ctx_table;
+
+    /* Number of channels */
+    unsigned int channel_count;
+};
+
+static struct smt_ctx smt_ctx;
+
+/*
+ * SCMI Transport API
+ */
+static int smt_get_secure(fwk_id_t channel_id, bool *secure)
+{
+    if (secure == NULL) {
+        assert(false);
```

Comment:
Please use the provided assert functions from framework `fwk_assert`

Reply:
ok